src/hooks/useInitialValues.test.ts

```diff
@@ -333,4 +334,239 @@ describe('useInitialValues', () => {
+      allFieldsKeys,
+    );
+
+    await act(async () => {
+      hook = renderHook(() =>
+        useInitialValues(
+          [...formFields],
+          encounter,
+          false,
+          {
+            encounter: encounter,
+            patient: testPatient,
+            location,
+            sessionMode: 'enter',
+            encounterDate: encounterDate,
+            setEncounterDate: jest.fn,
+            encounterProvider: '2c95f6f5-788e-4e73-9079-5626911231fa',
+            setEncounterProvider: jest.fn,
+            setEncounterLocation: jest.fn,
+            encounterRole: '',
+            setEncounterRole: jest.fn,
+          },
+          formFieldHandlers,
+        ),
+      );
+    });
+    const {
+      current: { initialValues, isBindingComplete },
+    } = hook.result;
+    expect(isBindingComplete).toBe(true);
+
+    const heightVal = initialValues['height'];
+    const weightVal = initialValues['weight'];
+
+    const calculatedBmi = helper.calcBMI(heightVal, weightVal);
+
+    expect(initialValues['height']).toBe(176);
+    expect(initialValues['weight']).toBe(56);
+    expect(initialValues['bmi']).toBe(calculatedBmi);
+  });
+
+  it('should return asynchronous calculated values for calculated fields in "edit" mode', async () => {
+    let hook = null;
+    const fieldWithCalculateExpression: FormField = {
+      label: 'Latest mother HIV status',
+      type: 'obs',
+      questionOptions: {
+        rendering: 'fixed-value',
+        concept: 'af7c1fe6-d669-414e-b066-e9733f0de7a8',
+        calculate: {
+          calculateExpression:
+            "resolve(api.getLatestObs(patient.id, '159427AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA', '2549af50-75c8-4aeb-87ca-4bb2cef6c69a'))?.valueCodeableConcept?.coding[0]?.code",
+        },
+      },
+      id: 'latest_mother_hiv_status',
+    };
+    allFormFields.push(fieldWithCalculateExpression);
+    await act(async () => {
+      hook = renderHook(() =>
+        useInitialValues(
+          [...allFormFields],
+          null,
+          false,
+          {
+            encounter: undefined,
+            patient: testPatient,
+            location,
+            sessionMode: 'enter',
+            encounterDate: encounterDate,
+            setEncounterDate: jest.fn,
+            encounterProvider: '2c95f6f5-788e-4e73-9079-5626911231fa',
+            setEncounterProvider: jest.fn,
+            setEncounterLocation: jest.fn,
+            encounterRole: '',
+            setEncounterRole: jest.fn,
+          },
+          formFieldHandlers,
+        ),
+      );
+    });
+    const {
+      current: { initialValues, isBindingComplete },
+    } = hook.result;
+    expect(isBindingComplete).toBe(true);
+    expect(initialValues['latest_mother_hiv_status']).toBe('664AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA');
+  });
+
+  it('should fall back to encounter value if the calculated expression result is null or undefined', async () => {
+    let hook = null;
+    let formFields: Array<FormField> = [
```

Comment:
This looks very similar to the other `formFields` variable. It would be easier to understand what the difference is if you defined this at the top level of the test file and then used it in each test file where it is needed, building new modified copies using the object spread syntax as necessary.